File: LearningProject/Models/ExamVariant.cs
namespace LearningProject.Models
{
public class ExamVariant
{
    public int Id { get; set; }
    public int ExamDefinitionId { get; set; }
    public required string Name { get; set; }

    // Navigation property for QuestionExam
    public ICollection<ExamQuestion>? ExamQuestions { get; set; }
    
    public ICollection<Result>? Results { get; set; }
    public ExamDefinition? ExamDefinition { get; set; }
}
}

File: client/src/app/core/models/ExamVariant.ts
import { ExamDefinition } from "./ExamDefinition";
import { ExamQuestion } from "./ExamQuestion";
import { Result } from "./Result";

export interface ExamVariant {
    id?: number;
    examDefinitionId?: number;
    variantName?: string | null;
    examQuestions?: ExamQuestion[] | null;
    results?: Result[] | null;
    examDefinition?: ExamDefinition | null;
  }
  